### PR TITLE
Add support for embedded surveys in post or page content on webkit.org

### PR DESCRIPTION
**Pull request by jdatapple**
#### 0ee248acc11659aa7ba747c89638068a5e5116b9
<pre>
Add support for embedded surveys in post or page content on webkit.org
<a href="https://bugs.webkit.org/show_bug.cgi?id=249342">https://bugs.webkit.org/show_bug.cgi?id=249342</a>

Reviewed by Timothy Hatcher.

* Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php: Added.
* Websites/webkit.org/wp-content/plugins/webkit-survey/results.php: Added.
* Websites/webkit.org/wp-content/plugins/webkit-survey/survey.php: Added.

Canonical link: <a href="https://commits.webkit.org/257891@main">https://commits.webkit.org/257891@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/e91f004be5c5d8a51df45f9220f10bd0928f4ebb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/6/builds/100339 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/9501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/33403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/109657 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/10412 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/38 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/92740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/107535 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106117 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/10412 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/33403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/92740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/10412 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/33403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/92740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/3243 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/33403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/3236 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/38 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/9357 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/33403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/5052 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2801 "Built successfully and passed tests") | | | |

### CODE REVIEW

File: Websites/webkit.org/wp-content/plugins/webkit-survey/survey.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            <input type="text" name="other[<?php echo esc_attr($id); ?>]" maxlength="144">
+        <?php endif; ?>
+        </label></li>
+    <?php endforeach; ?>
+    </ul>
+<?php endforeach; ?>
+<p class="button-align-right">
+    <input type="submit" name="Submit" value="Submit" class="submit-button">
+</p>
+</form>
```
Comment (xeenon) [inline Websites/webkit.org/wp-content/plugins/webkit-survey/survey.php]: Should add a newline at the end of the file.

**Review by xeenon — APPROVED** [review]

File: Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    }
+    
+    private static function voter_hash() {
+        $ip = filter_input(INPUT_SERVER, 'REMOTE_ADDR');
+        $ua = filter_input(INPUT_SERVER, 'HTTP_USER_AGENT');
+        return sha1($ip . $ua);
+    }
+}
+
+WebKit_Survey::init();
```
Comment (xeenon) [inline Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php]: Newline here too.

File: Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+            $ui_template = $results_template;
+        
+        ob_start();
+        if ($atts['visible-results'] == "true")
+            include($results_template);
+        include_once($ui_template);
+        return ob_get_clean();
+    }
+
+    public function add_survey_boxes () {
```
Comment (xeenon) [inline Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php]: No space before `()`

File: Websites/webkit.org/wp-content/plugins/webkit-survey/survey.php (hunk 1 above)
Comment (jdatapple) [reply Websites/webkit.org/wp-content/plugins/webkit-survey/survey.php]: Addressed in 2619fb8

File: Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php (hunk 2 above)
Comment (jdatapple) [reply Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php]: Addressed in 2619fb8

File: Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php (hunk 3 above)
Comment (jdatapple) [reply Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php]: Addressed in e91f004

**Pull request merged**